### PR TITLE
Faster REPT Serde

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
Implementation of Native Arrow Format on Row Entity Phi Tensor.

## Affected Dependencies
REPT Tesor

## How has this been tested?
Tests to be added.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/core/tensor/lazy_repeat_array.py [hunk 1]
```diff
@@ -83,32 +102,59 @@ def __pow__(self, exponent: int) -> lazyrepeatarray:
...
         return self == other
+
+    @property
+    def dtype(self) -> np.dtype:
+        return self.data.dtype
+
+    def astype(self, np_type: np.dtype) -> lazyrepeatarray:
+        return self.__class__(self.data.astype(np_type), self.shape)
+
+    def to_numpy(self, shape: Optional[Tuple[int, ...]] = None) -> np.ndarray:
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/tensor/lazy_repeat_array.py:146]: Nice :+1:

File: packages/syft/src/syft/core/tensor/lazy_repeat_array.py [hunk 2]
```diff
@@ -83,32 +102,59 @@ def __pow__(self, exponent: int) -> lazyrepeatarray:
...
+            return self.to_numpy().sum()
         else:
             raise Exception("not sure how to do this yet")
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Any) -> lazyrepeatarray:  # type: ignore
         if isinstance(other, lazyrepeatarray):
-            return self.data == other.data and self.shape == other.shape
+            if self.shape == other.shape:
+                return lazyrepeatarray(data=self.data == other.data, shape=self.shape)
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/tensor/lazy_repeat_array.py:122]: Q: Should we not return a boolean , here why are we wrapping in a lazy repeat array.

File: packages/syft/tests/syft/core/tensor/adp/ndim_entity_phi_test.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    # ser = sy.serialize(tensor1, to_bytes=True)
+    # de = sy.deserialize(ser, from_bytes=True)
+
+    assert de == tensor1
+    assert (de.child == tensor1.child).all()
+    assert (de.min_vals == tensor1.min_vals).all()
+    assert (de.max_vals == tensor1.max_vals).all()
+    assert de.entities == tensor1.entities
+
+    assert np.shares_memory(tensor1.child, tensor1.child)
```
Comment (rasswanth-s) [inline packages/syft/tests/syft/core/tensor/adp/ndim_entity_phi_test.py:214]: Wonderful the way you test zero copy.

Comment (iamtrask) [reply packages/syft/tests/syft/core/tensor/adp/ndim_entity_phi_test.py:211]: +1

**Pull request merged**